### PR TITLE
Add number of patches to file metadata

### LINKED ISSUE #176: TwoPoint metadata should include the number of patches it uses for jackknifing
Currently the number of jackknife patches used by TwoPoint is not saved in the metadata, it probably should be.

### PR DESCRIPTION
**Pull request by joezuntz**
Saves the number of patches as metadata in the sacc file output when present.

Closes #176

### CODE REVIEW

File: txpipe/twopoint.py [hunk 1]
```diff
@@ -414,7 +414,7 @@ def call_treecorr(self, i, j, k):
         sys.stdout.flush()
         return result
 
-    def prepare_patches(self, calcs):
+    def prepare_patches(self, calcs, meta):
```
Comment (empEvil) [inline txpipe/twopoint.py:417]: The documentation has not been updated

**Review by empEvil — CHANGES_REQUESTED** [review]
Looks good, its a minor issue that we should update the documentation for the changes!

File: txpipe/twopoint.py (hunk 1 above)
Comment (joezuntz) [reply txpipe/twopoint.py:417]: Thanks, I've fixed!

File: txpipe/utils/patches.py [hunk 2]
```diff
@@ -241,23 +241,24 @@ def run(cls, cat, chunk_rows, comm=None):
                 print(
                     f"Catalog {cat.file_name} does not have a patch directory set, so not making patches."
                 )
-            return
+            return 0
```
Comment (empEvil) [inline txpipe/utils/patches.py:244]: It should probably also be mentioned in the documentation for this class that it now returns a scalar value!

**Review by empEvil — APPROVED** [review]
Great!

**Pull request merged**